src/Microsoft.Framework.DesignTimeHost/Models/OutgoingMessages/DiagnosticsMessage.cs

```diff
@@ -10,30 +10,91 @@ namespace Microsoft.Framework.DesignTimeHost.Models.OutgoingMessages
+                .Select(msg => new DiagnosticsInfo
+                {
+                    Path = msg.SourceFilePath,
+                    Line = msg.StartLine,
+                    Column = msg.StartColumn,
+                    Message = msg.Message,
+                    FormattedMessage = msg.FormattedMessage
+                });
+
+            Framework = frameworkData;
+            CompilationDiagnostics = compilationMessages;
+        }
+
+        public DiagnosticsMessage(IEnumerable<DiagnosticsInfo> errors, IEnumerable<DiagnosticsInfo> warnings, FrameworkData framework)
+        {
+            Errors = errors ?? Enumerable.Empty<DiagnosticsInfo>();
+            Warnings = warnings ?? Enumerable.Empty<DiagnosticsInfo>();
+            Framework = framework;
+        }
 
         [JsonIgnore]
-        public IList<ICompilationMessage> Diagnostics { get; set; }
+        public IList<ICompilationMessage> CompilationDiagnostics { get; }
 
-        public IEnumerable<string> Errors => Diagnostics.Where(d => d.Severity == CompilationMessageSeverity.Error)
-                                                        .Select(d => d.FormattedMessage);
+        public FrameworkData Framework { get; }
 
-        public IEnumerable<string> Warnings => Diagnostics.Where(d => d.Severity == CompilationMessageSeverity.Warning)
-                                                        .Select(d => d.FormattedMessage);
+        public IEnumerable<DiagnosticsInfo> Errors { get; }
+
+        public IEnumerable<DiagnosticsInfo> Warnings { get; }
 
         public override bool Equals(object obj)
         {
             var other = obj as DiagnosticsMessage;
 
             return other != null &&
-                 Enumerable.SequenceEqual(Warnings, other.Warnings) &&
-                 Enumerable.SequenceEqual(Errors, other.Errors);
+                Enumerable.SequenceEqual(Errors, other.Errors) &&
+                Enumerable.SequenceEqual(Warnings, other.Warnings) &&
+                Framework == other.Framework;
+        }
+
+        public override int GetHashCode()
+        {
+            return base.GetHashCode();
+        }
+    }
+
+    public class DiagnosticsInfo
+    {
+        public string Path { get; set; }
+
+        public int Line { get; set; }
+
+        public int Column { get; set; }
+
+        public string Message { get; set; }
+
+        public string FormattedMessage { get; set; }
+
+        public override bool Equals(object obj)
+        {
+            var other = obj as DiagnosticsInfo;
+
+            return other != null &&
+                 other.Line == Line && 
+                 other.Column == Column && 
+                 other.Message == Message && 
+                 other.FormattedMessage == FormattedMessage;
         }
 
         public override int GetHashCode()
         {
-            // These objects are currently POCOs and we're overriding equals
-            // so that things like Enumerable.SequenceEqual just work.
```

Comment:
Yeah, I thought it's unnecessary but it's needed actually. If `Equals` is overridden, the `GetHashCode` must be overriden. [CA2218(https://msdn.microsoft.com/en-us/library/ms182358.aspx)]